src/main/scala/de/frosner/ddq/constraints/TypeConversionConstraint.scala

```diff
@@ -3,21 +3,49 @@ package de.frosner.ddq.constraints
 import org.apache.spark.sql.types.DataType
 import org.apache.spark.sql.{Column, DataFrame}
 
+import scala.util.Try
+
 case class TypeConversionConstraint(columnName: String,
                                     convertedType: DataType) extends Constraint {
 
   val fun = (df: DataFrame) => {
     val originalColumn = new Column(columnName)
     val castedColumnName = columnName + "_casted"
-    val casted = df.select(originalColumn, originalColumn.cast(convertedType).as(castedColumnName))
-    val failedCastsCount = casted.filter(new Column(castedColumnName).isNull && originalColumn.isNotNull).count
-    val originalType = df.schema.find(_.name == columnName).get.dataType
+    val maybeCasted = Try(df.select(originalColumn, originalColumn.cast(convertedType).as(castedColumnName)))
+    val maybeFailedCastsCount = maybeCasted.map(_.filter(new Column(castedColumnName).isNull && originalColumn.isNotNull).count)
+    val maybeOriginalType = maybeFailedCastsCount.map(_ => df.schema.find(_.name == columnName).get.dataType)
     TypeConversionConstraintResult(
-      this,
-      originalType,
-      failedCastsCount,
-      if (failedCastsCount == 0) ConstraintSuccess else ConstraintFailure
+      constraint = this,
+      data = maybeOriginalType.toOption.map(originalType => TypeConversionConstraintResultData(
+        originalType = originalType,
+        failedRows = maybeFailedCastsCount.get
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/frank_7/drunken-data-quality/file/2305054522/issues/source?bid=3033737&fileBranchId=3196266#l21)